sound/soc/sof/pcm.c

```diff
@@ -105,9 +105,8 @@ int sof_pcm_dsp_pcm_free(struct snd_pcm_substream *substream, struct snd_sof_dev
 	return ret;
 }
 
-static int sof_pcm_setup_connected_widgets(struct snd_sof_dev *sdev,
-					   struct snd_soc_pcm_runtime *rtd,
-					   struct snd_sof_pcm *spcm, int dir)
+int sof_pcm_setup_connected_widgets(struct snd_sof_dev *sdev, struct snd_soc_pcm_runtime *rtd,
+				    struct snd_sof_pcm *spcm, int dir)
```

Comment:
The `sof_widget_list_setup()` call in this function is ipc3 specific, which kind of makes this function ipc3 specific, no?

Reply:
sof_widget_list_setup() is not IPC-specific anymore? I changed it in this PR. The 2nd patch

Reply:
I also moved this to pcm.c because IPC4 will call it too